src/aichat/chatapp/load_df.py

```diff
@@ -11,62 +17,93 @@ def load_df(path=DATA_DIR):
     for column in CSV_COLUMNS:
         df_clean[column] = df[column]
     df_clean.to_csv("Export.csv", index=None)
     df_clean = df_clean.reset_index(drop=True)
-    df_clean = df_clean.fillna('')
+    df_clean = df_clean.fillna('root')
     return df_clean
 
 
-def states_to_list(state, path=DATA_DIR):
-    state_list = []
-    df = load_df(path)
-    if state == 'source_state':
-        state_list = list(set(df.source_state))
-    elif state == 'dest_state':
-        state_list = list(set(df.dest_state))
-    elif state == 'response':
-        state_list = list(df.response)
-    elif state == 'trigger':
-        state_list = list(df.trigger)
-    return state_list
+DF = load_df()
 
 
-def nodes_to_dict(path=DATA_DIR):
-    state_list = states_to_list('source_state')
-    dest_state = states_to_list('dest_state')
-    node_list = state_list + dest_state
-    node_list = list(set(node_list))
+def nodes_to_list(path=DATA_DIR):
+    node_list = list(set(DF.dest_state.fillna('root')))
     dict_list = []
-    for node in range(len(node_list)):
-        dict_list.append({'name:': node_list[node], 'id': 'node' + str(node)})
+    node_index = 0
+    for node in node_list:
+        # if '*' not in node and '?' not in node and '|' not in node:
+        dict_list.append(
+            {'name': node, 'id': 'node' + str(node_index)})
+        node_index = node_index + 1
     return dict_list
 
 
-def links_to_list(path=DATA_DIR):
-    trigger_list = states_to_list('trigger')
-    response_list = states_to_list('response')
-    return trigger_list, response_list
+def links_to_list(path=DATA_DIR, value=1):
+    # links = []
+    # source_list = DF.source_state
+    # dest_list = DF.dest_state
+    # nodes = nodes_to_list()
+    # nodes_index = {}
+    # for node_name in nodes:
+    #     nodes_index[node_name['name']] = node_name['id'].replace('node', '')
+    # for i, source in enumerate(source_list):
+    #     patt = pattern.expand_globstar(source)
+    #     for j, dest in enumerate(dest_list):
+    #         match = regex.match(patt, dest)
+    #         if match:
+    #             # if int(nodes_index[source]) == int(nodes_index[dest]):
+    #             #     continue
+    #             links.append({'source': int(nodes_index[source]),
+    #                           'target': int(nodes_index[dest]),
+    #                           'command': DF.trigger[j],
+    #                           'response': DF.response[j],
+    #                           'value': value})
+    # return links
+    links = []
+    node_names = list(set(DF.dest_state))  # list of node names (states)
+    # print(nodes_index)
+    for i, source_pattern in enumerate(DF.source_state.values):
+        patt = pattern.expand_globstar(source_pattern)
+        for j, name in enumerate(node_names):
+            if ('{' in patt or '[' in patt) and regex.match(patt, name):
+                print(str(j) + source_pattern + ',' + name)
```

Comment:
Consider using the python logger instead of print statements. https://docs.python.org/2/library/logging.html